src/components/ConfigEditor.tsx

```diff
@@ -68,6 +68,30 @@ export function ConfigEditor(props: Props) {
     });
   };
 
+  const [queryTimeoutValidationError, setQueryTimeoutValidationError] = useState<string | null>(null);
+  const onChangeQueryTimeoutSec = (event: ChangeEvent<HTMLInputElement>) => {
+    const validateQueryTimeout = (value: string | undefined): boolean => {
+      if (value != null && value.length > 0) {
+        const queryTimeoutSec = +value;
+        if (Number.isNaN(queryTimeoutSec) || queryTimeoutSec <= 0) {
+          setQueryTimeoutValidationError(`Invalid query timeout (${value}), must be a positive integer`);
+          return false;
+        }
+      }
+      setQueryTimeoutValidationError(null);
+      return true;
+    };
+    if (validateQueryTimeout(event.target.value)) {
+      onOptionsChange({
+        ...options,
+        jsonData: {
+          ...options.jsonData,
+          queryTimeoutSec: event.target.value?.length > 0 ? +event.target.value : undefined,
```

Comment:
nit: do we need this `event.target.value?.length > 0` check if `validateQueryTimeout` already performs the check?

Reply:
oh wait ignore, it's still valid if it's empty, nvm!!

Reply:
Yep, exactly!